db/migrate/20160321160404_add_middleware_event_streams.rb

```diff
@@ -0,0 +1,7 @@
+class AddMiddlewareEventStreams < ActiveRecord::Migration[5.0]
+  def change
+    add_column :event_streams, :middleware_type, :string   # relevant type (entity class name)
+    add_column :event_streams, :middleware_id, :bigint     # MIQ entity id
```

Comment:
@Fryguy ems_id reflects the id of the relevant ext_management_system.  I don't actually see an event_streams.ems_type field.  The fields you ask about are not holding the ems but rather the relevant resource.  So, for example, for an event relevant to a middleware server then middleware_type='MiddlewareServer' and middleware_id would get set to the relevant id for the middleware_servers table.